packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -238,7 +238,13 @@ async function nodeImport(
         return nodeResolve(id, parent, isMain, options)
       }
       if (parent) {
-        return viteResolve(id, parent.id)
+        let resolved = viteResolve(id, parent.id)
+        if (resolved) {
+          // hookNodeResolve must use platform-specific path.normalize
+          // to be compatible with dynamicImport (#6080)
+          resolved = path.normalize(resolved)
```

Comment:
There is a `normalizePath` function from util, can we use it?

https://github.com/vitejs/vite/blob/main/packages/vite/src/node/utils.ts#L145-L147

Reply:
No, the `normalizePath` in util calls `path.posix` and always returns path with forward slashes which is not what's needed here.

This place is special, this is `hookNodeResolve` which is a [replacement](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/ssrRequireHook.ts#L72) for internal node module resolution so it should return path in the same format as internal node module resolution does, and that uses native format - backslashes on windows and forward slashes on unix, so node `path.normalize` is used here. Otherwise the problem in #6080 happens - the same module is loaded twice, due to its path being normalized in two different ways, one when the hook is in place and the other one when it's not.

Reply:
I see. Sorry for the noise.